kernelci/lava.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2019 Linaro Limited
+# Author: Dan Rue <dan.rue@linaro.org>
+#
+# This module is free software; you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation; either version 2.1 of the License, or (at your option)
+# any later version.
+#
+# This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+
+def get_device_type_by_name(name, device_types, aliases=[]):
+    """
+        Return a device type named name. In the case of an alias, resolve the
+        alias to a device_type
+
+        Example:
+        IN:
+            name = "x15"
+            device_types = [
+              {'busy': 1, 'idle': 0, 'name': 'am57xx-beagle-x15', 'offline': 0},
+              {'busy': 1, 'idle': 0, 'name': 'beaglebone-black', 'offline': 0},
+              {'busy': 0, 'idle': 1, 'name': 'docker', 'offline': 0},
+              {'busy': 0, 'idle': 1, 'name': 'qemu', 'offline': 0}
+            ]
+            aliases = [
+              {'name': 'x15', 'device_type': 'am57xx-beagle-x15'}
+            ]
+        OUT:
+            {'busy': 1, 'idle': 0, 'name': 'am57xx-beagle-x15', 'offline': 0}
+
+    """
+    for device_type in device_types:
+        if device_type["name"] == name:
+            return device_type
+    for alias in aliases:
+        if alias["name"] == name:
```

Comment:
Do we actually have to care about how the alias is called in the lab?  Is it not enough to check that there is one alias with a `device_type` that matches the `name`?  Here we're comparing `am57xx-beagle-x15` with `x15` so it looks like something isn't quite right.  (see my previous comment as I'm not sure which way round this should be)

Reply:
OK thanks for making the example in the docstring more like a typical KernelCI case.  That makes sense now, it's not just about the device type name as a string but the rest of the dictionary is needed to submit jobs to online devices etc...